docs/testplan/transceiver_onboarding_test_plan.md

```diff
@@ -0,0 +1,357 @@
+# Transceiver Onboarding Test Plan
+
+### Scope
+
+The scope of this test plan is to layout a framework of tests to ensure feature parity of new transceivers being onboarded to SONiC with the existing feature support. The test plan is planned to be automated.
+
+The below features/properties of the transceiver are to be tested in this test plan.
+
+- Verify link behavior with shut/no shut commands
+- Verify link behavior with process crash scenarios
+- Verify transceiver specific fields (such as Vendor name, part number and serial number) have expected value using CLI command.
+- Check the fimware version is returned correctly and is one of the acceptable values suggested by vendor. The major and minor number can be qualified by a regex match with a predefined pattern
+- Verify DOM data is read correctly and is in acceptable range.
+- Check if any unexpected flags (such as LOS, LOL, DOM related warnings) are not set
+- Ensure firmware download, upgrade and downgrade works fine.
+- Verify that number of frames received is equal to number of frames sent in both northbound/southbound traffic flows for a predefined set period of time.
+- Verify traffic at line rate
```

Comment:
traffic at line rate can be affected by other configurations, such as buffer, packet length, etc.  Do want to define a test with a set of specific configurations?

Reply:
I have now removed the traffic test portion from the HLD. This was done to keep the test plan restricted to testing transceiver specific features.
A generic set of traffic tests should be run independently of the current test plan.